src/mz390.java

```diff
@@ -12315,21 +12317,20 @@ private void exec_pc_x2b(){
 		}
 		put_setc_stack_var();
     }
-    private void exec_pc_x2c(){
-    	/*
-    	 * convert hex string to char string
-    	 */
-    	check_setc_quotes(1); // RPI 1139
-    	setc_value1 = get_setc_stack_value();
-    	seta_value = Integer.valueOf(setc_value1,16);
-		setc_value = ""
-			       + (char)tz390.ebcdic_to_ascii[seta_value >>> 24]
-			       + (char)tz390.ebcdic_to_ascii[seta_value >>> 16 & 0xff]         
-			       + (char)tz390.ebcdic_to_ascii[seta_value >>> 8  & 0xff]
-			       + (char)tz390.ebcdic_to_ascii[seta_value        & 0xff]					                               
-			       ;
+	private void exec_pc_x2c(){
+		/*
+		 * convert hex string to char string
+		 */
+		check_setc_quotes(1); // RPI 1139
+		setc_value1 = get_setc_stack_value();
+		StringBuilder stb = new StringBuilder("");
+		for (int i = 0; i < setc_value1.length(); i += 2) {
+			String str = setc_value1.substring(i, i + 2);
+			stb.append((char) Integer.parseInt(str, 16));
```

Comment:
The above line adds an ASCII character to stb. Should be an EBCDIC character. The line should be
```
stb.append((char) tz390.ebcdic_to_ascii[Integer.parseInt(str, 16) & 0xff]);
```
Attached file contains output from test program that was assembled with mz390.java modified as suggested. 
[TX2C.TXT](https://github.com/z390development/z390/files/9639451/TX2C.TXT)

Reply:
Already put this one in an email. Needs to be here.

The suggested fix has a problem when the tz390.ebcdic_to_ascii array entry has a (byte) value exceeding X'7F'. Values X'80' -- X'FF' are negative. The "char" cast adds X'FF' on the left of such a value. The line must be updated to
```
    stb.append((char)(((int)tz390.ebcdic_to_ascii[Integer.parseInt(str, 16)]) & 0xff));
```
Sorry about that.